draft-ietf-quic-multipath.md

```diff
@@ -1416,34 +1398,46 @@ Sequence Number:
 
 ## MAX_PATHS frames {#max-paths-frame}
 
-A MAX_PATHS frame (type=0x15228c0b) informs the peer of the cumulative number of paths
-it is permitted to open.
+A MAX_PATHS frame (type=0x15228c0b) informs the peer of largest Path ID that
+is permitted to use on any path.
 
 MAX_PATHS frames are formatted as shown in {{fig-max-paths-frame-format}}.
 
 ~~~
 MAX_PATHS Frame {
   Type (i) = 0x15228c0b,
-  Maximum Paths (i),
+  Maximum Path Identifier (i),
 }
 ~~~
 {: #fig-max-paths-frame-format title="MAX_PATHS Frame Format"}
 
 MAX_PATHS frames contain the following field:
 
 Maximum Path Identifier:
-: A count of the cumulative number of path that can be opened
-  over the lifetime of the connection. This value cannot exceed 2^32-1, as it is not
-  possible to encode Path IDs larger than 2^32-1. Receipt of a frame that permits
-  opening of a path with Path ID larger than this limit MUST be treated
-  as a connection error of type FRAME_ENCODING_ERROR.
+: The largest Path ID that can be used for this connection.
+  This value cannot exceed 2^32-1, as it is not
+  possible to encode Path IDs larger than 2^32-1 as variable integer.
+
+Receipt of PATH_AVAILABLE, PATH_STANDBY, PATH_ABANDON or MP_ACK frames
+that uses a Path ID that is larger than the announced Path ID
+MUST be treated as a connection error of type FRAME_ENCODING_ERROR.
 
 Loss or reordering can cause an endpoint to receive a MAX_PATHS frame with
-a lower path limit than was previously received. MAX_PATHS frames that
-do not increase the path limit MUST be ignored.
+a smaller Path ID than was previously received. MAX_PATHS frames that
+do not announce a larger Path ID than previously received MUST be ignored.
+
+Endpoints SHOULD NOT issue new connection IDs which have path identifiers larger than
+the Path ID announced in the Maximum Path Identifier field in the
+MP_MAX_PATHS frame {{max-paths-frame}}.
+If no MAX_PATHS frame was received yet, the Maximum Path Identifier
+corresponds to the value of initial_max_paths transport parameter.
+
+An endpoint MUST NOT initiate a path with a Path ID larger than the Maximum Path Identifier value.
+An endpoint MUST terminate the a connection with an error of type MP_PROTOCOL_VIOLATION
+if a peer opens a path with a larger Path ID than permitted.
 
-An endpoint MUST NOT initiate a path with a Path ID higher than the Maximum Paths value.
-An endpoint MUST terminate the a connection with an error of type MP_PROTOCOL_VIOLATION if a peer opens more paths than was permitted.
+When there are not enough unused path identifiers, endpoints can
+send a MAX_PATHS frame to inform the peer that new path identifiers are available.
```

Comment:
I also proposed to remove this sentence for now. First it is unclear what "not enough" means and second we have two open issues to discuss this further (see #325 and #338).

```suggestion
```

Reply:
Oh, we need to keep the mechanism for updating the Path ID limit. 
"Not enough" means all the available Path IDs issued by transport parameter and previous MAX_PATHS(_ID) frames are all used in current paths. If endpoints wants to create new paths(also need to issue new CIDs associated with these Path IDs) at this time, they need to use MAX_PATHS(_ID) frame to inform the peer that new used Path ID resources are allowed.

Reply:
Okay looking at 5.1.1 ([Issuing Connection IDs](https://datatracker.ietf.org/doc/html/rfc9000#name-issuing-connection-ids)) in RFC9000, I think we should not state these rules normatively again, we should just say these rules should now be applied per path.